pkg/controllers/allocation/controller.go

```diff
@@ -77,7 +77,11 @@ func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface
 
 // Reconcile executes an allocation control loop for the resource
 func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
-	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named("Allocation"))
+	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named(fmt.Sprintf("allocation.provisioner/%s", req.Name)))
+	logging.FromContext(ctx).Infof("Starting provisioning loop")
+	defer func() {
+		logging.FromContext(ctx).Infof("Watching for pod events")
```

Comment:
Just curious, I'm not sure I can figure out why this is in a `defer`?

Reply:
This prints at the end of reconciliation. If the controller is hanging, the last log line will be "Watching for pod events". I could change this to something like "Waiting for pod events". The idea is to reassure users that karpenter is working even if its hanging.